system/doc/reference_manual/expressions.xml

```diff
@@ -1985,29 +1985,31 @@ KeyPattern := ValuePattern &lt;- MapExpression</pre>
     marker="stdlib:maps#iterator/2">maps:iterator/2</seemfa>.</p>
 
     <p>A <em>filter</em> is an expression that evaluates to
-    <c>true</c> or <c>false</c>, or a
-    <seeguide marker="#guard_expressions">guard expression</seeguide>.
-    If the filter is not a guard expression and evaluates
-    to a non-Boolean value <c>Val</c>, an exception
-    <c>{bad_filter, Val}</c> is triggered at runtime.</p>
+    <c>true</c> or <c>false</c>.</p>
 
     <p>The variables in the generator patterns shadow previously bound variables,
     including variables bound in a previous generator pattern.</p>
 
-    <p>A list comprehension returns a list, where the lists elements are the
+    <p>Variables bound in a generator expression is not visible outside the expression:</p>
+
+    <pre>
+1> <input>[{E,L} || E &lt;- L=[1,2,3]].</input>
+* 1:5: variable 'L' is unbound</pre>
+
+    <p>A <strong>list comprehension</strong> returns a list, where the lists elements are the
```

Comment:
I think there is a minor misspelling here, but I could be wrong.

I think you either want to say `where the list's elements` making clear that the list is in possession of the elements, or you want to say `where the list elements`. The current `where the lists elements` may not be grammatically correct, but I am not 100% sure...